internal/testrunner/runners/system/tester.go

```diff
@@ -897,73 +918,13 @@ func (r *tester) prepareScenario(ctx context.Context, config *testConfig, svcInf
 		policy = policyCurrent
 	}
 
-	agentDeployed, agentInfo, err := r.setupAgent(ctx, config, serviceStateData, policy, scenario.pkgManifest.Agent)
+	agentDeployed, agentInfo, err := r.setupAgent(ctx, config, serviceStateData, policy, r.pkgManifest.Agent)
 	if err != nil {
 		return nil, err
 	}
 
 	scenario.agent = agentDeployed
 
-	service, svcInfo, err := r.setupService(ctx, config, serviceOptions, svcInfo, agentInfo, agentDeployed, policy, serviceStateData)
```

Comment:
Delay setting up service, after setting the handlers related to the Agent
- shutdownAgentHandler (this is set within `setupAgent` method)
- removeAgentHandler
- resetAgentPolicyHandler
- resetAgentLogLevelHandler

Reply:
It cannot be done this change.
In `servicedeployer` package, there are some agents created there (custom agents and K8s agents). And therefore, it is required to call `checkEnrolledAgents` after `setupService`.

At least, until it is kept support for those scenarios using Elastic Agent from stack.